src/smc-webapp/library.cjsx

```diff
@@ -0,0 +1,319 @@
+            maxWidth      : '100%'
+            border        : "1px solid #{COLORS.GRAY_L}"
+            boxShadow     : "2px 2px 1px #{COLORS.GRAY_LL}"
+            borderRadius  : '5px'
+
+        return <img src={img_path} style={img_style} onLoad={=> @setState(show_thumb:true)} />
+
+
+    details: ->
+        return null if (not @props.library_selected?) or (not @state.metadata?)
+        # example:
+        # {"title":"Data science Python notebooks","id":"doc-6","license":"a20",
+        # "src":"/ext/library/cocalc-examples/data-science-ipython-notebooks/",
+        # "description":"Data science Python notebooks: Deep learning ...\n"}
+        doc   = @props.library_selected
+        meta  = @state.metadata
+        style =
+            maxHeight  : HEIGHT
+            overflow   : 'auto'
+
+        # this tells the user additional information for specific tags (like, pick the right kernel...)
+        tag_extra_info = []
+        for tag in doc.tags ? []
+            info = @state.metadata.tags[tag].info
+            tag_extra_info.push(info) if info
+
+        <div style={style}>
+            <h5 style={marginTop: '0px'}>
+                <strong>{doc.title ? doc.id}</strong>
+                {" by #{doc.author}" if doc.author?}
+            </h5>
+            {
+                if doc.description?
+                    <p style={color: COLORS.GRAY_D}>
+                        <Markdown value={doc.description} />
+                    </p>
+            }
+            {
+                if doc.website?
+                    website_style =
+                        whiteSpace    : 'nowrap'
+                        overflow      : 'hidden'
+                        textOverflow  : 'ellipsis'
+                    <p style={color: COLORS.GRAY_D}>
+                        Website: <a style={website_style} target='_blank' href={doc.website}>{doc.website}</a>
+                    </p>
+            }
+            {
+                if doc.license?
+                    <p style={color: COLORS.GRAY_D}>
+                        License: {meta.licenses[doc.license] ? doc.license}
+                    </p>
+            }
+            {
+                if doc.tags?
+                    tags = ((meta.tags[t].name ? t) for t in doc.tags)
+                    <p style={color: COLORS.GRAY_D}>
+                        Tags: {tags.join(', ')}
+                    </p>
+            }
+            {#<p style={color: '#666'}>copies <code>{@props.library_selected.src}</code> into <code>{@target_path()}</code></p>}
+            {
+                if tag_extra_info.length > 0
+                    info = tag_extra_info.join(' ')
+                    <p style={color: COLORS.GRAY_D}>
+                        <Icon name='exclamation-triangle' style={color:COLORS.YELL_L} /> {info}
+                    </p>
+            }
+            <Button
+                bsStyle  = "success"
+                onClick  = {=> @copy()}
+                disabled = {@state.copy}
```

Comment:
This copy state should go into the store so that if a user goes navigates elsewhere the copying status will still be correct.